src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -470,6 +489,43 @@ MethodSymbol getTopLevelMethod(MethodSymbol method)
                 }
                 return null;
             }
+
+            // Returns true if the nullable state from the operand of the conversion
+            // can be used as is, and we can create a slot from the conversion.
+            bool isSupportedConversion(Conversion conversion, BoundExpression operandOpt)
+            {
+                // https://github.com/dotnet/roslyn/issues/32599: Allow implicit and explicit
+                // conversions where the nullable state of the operand remains valid.
+                switch (conversion.Kind)
+                {
+                    case ConversionKind.Identity:
+                    case ConversionKind.DefaultOrNullLiteral:
+                        return true;
+                    case ConversionKind.ImplicitReference:
+                        return operandOpt?.IsLiteralNullOrDefault() == true;
+                    case ConversionKind.ImplicitTupleLiteral:
+                        {
+                            var arguments = ((BoundConvertedTupleLiteral)operandOpt).Arguments;
+                            var conversions = conversion.UnderlyingConversions;
+                            for (int i = 0; i < arguments.Length; i++)
+                            {
+                                // https://github.com/dotnet/roslyn/issues/32600: Copy nullable
```

Comment:
>// https://github.com/dotnet/roslyn/issues/32600: Copy nullable [](start = 32, length = 63)

If possible, can you include tests for this yet-unhandled scenario (and the one a few lines below) to illustrate the bug/problem?

Reply:
I've added example tests to the bugs. I'd prefer to leave the examples there for now, rather than adding tests here, since the set of tests needed when the bugs are fixed might be organized differently.

---
In reply to: [249171645](https://github.com/dotnet/roslyn/pull/32612#discussion_r249171645) [](ancestors = 249171645)